app/controllers/declaration/AuthorisationProcedureCodeChoiceController.scala

```diff
@@ -57,12 +59,23 @@ class AuthorisationProcedureCodeChoiceController @Inject()(
       .bindFromRequest()
       .fold(
         errors => Future.successful(BadRequest(authorisationProcedureCodeChoice(errors, mode))),
-        updateCache(_).map(_ => navigator.continueTo(mode, routes.DeclarationHolderSummaryController.displayPage))
+        updateCache(_).map(exportsDeclarationUpdated => navigator.continueTo(mode, nextPage(exportsDeclarationUpdated)))
       )
   }
 
   private def updateCache(
     choice: AuthorisationProcedureCodeChoice
   )(implicit request: JourneyRequest[AnyContent]): Future[Option[ExportsDeclaration]] =
     updateExportsDeclarationSyncDirect(_.updateAuthorisationProcedureCodeChoice(choice))
+
+  private def nextPage(exportsDeclarationOpt: Option[ExportsDeclaration]): Mode => Call =
+    (for {
+      exportsDeclaration <- exportsDeclarationOpt
+      authProcedureCodeChoice <- exportsDeclaration.parties.authorisationProcedureCodeChoice.map(_.code)
+
+    } yield (exportsDeclaration.`type`, exportsDeclaration.additionalDeclarationType, authProcedureCodeChoice)) match {
+      case Some((STANDARD, Some(STANDARD_PRE_LODGED), Code1040 | CodeOther)) => routes.DeclarationHolderRequiredController.displayPage
```

Comment:
What if holders is already populated, should it not go to the summary page here?